dd-java-agent/instrumentation/aws-lambda-handler/src/main/java/datadog/trace/instrumentation/aws/v1/lambda/LambdaHandlerInstrumentation.java

```
@@ -108,8 +100,9 @@ static AgentScope enter(
     static void exit(
         @Origin String method,
         @Enter final AgentScope scope,
-        @Advice.Return final Object result,
+        @Advice.Return(typing = DYNAMIC) final Object result,
```

https://javadoc.io/static/net.bytebuddy/byte-buddy/1.10.5/net/bytebuddy/asm/Advice.Return.html


> Indicates that the annotated parameter should be mapped to the return value of the instrumented method. If the instrumented method terminates exceptionally, the type's default value is assigned to the parameter, i.e. 0 for numeric types and null for reference types. If the return type is void, the annotated value is null if and only if typing() is set to Assigner.Typing#DYNAMIC.

The "handleRequest" method on RequestStreamHandler has a void return type, which was causing this to silently fail to apply the advice for that type.

---

What am I missing here? Is there one method that returns something and one that doesn't? If there isn't then shouldn't the `@Advice.Return ...` just be removed?

---

There are two interfaces, `RequestStreamHandler` and `RequestHandler`. On `RequestHandler` the handleRequest method returns a value, (the interface is generic), and `RequestStreamHandler` returns void. We want to capture the return value for the  RequestHandler case

---

Thanks @DarcyRaynerDD, makes perfect sense then.